uaclient/files/notices.py

```diff
@@ -0,0 +1,244 @@
+        """Deletes a notice file.
+
+        :param notice_details: Holds details concerning the notice file.
+        """
+        if root_mode:
+            label = notice_details.label
+            file_order_id = notice_details.order_id
+            is_notice_permanent = notice_details.is_permanent
+            self.directory = (
+                defaults.NOTICES_PERMANENT_DIRECTORY
+                if is_notice_permanent
+                else defaults.NOTICES_TEMPORARY_DIRECTORY
+            )
+            filename = "{}-{}".format(file_order_id, label)
+            notice_files = self.get_all_notice_files()
+            if notice_files:
+                label_file = [
+                    file for file in notice_files if file == filename
+                ]
+                if len(label_file) == 1:
+                    system.remove_file(
+                        os.path.join(
+                            self.directory, label_file[0]  # type:ignore
+                        )
+                    )
+        else:
+            event.warning("Trying to remove a notice as non-root user")
+
+    def get_all_notice_files(
+        self, notice_dir: Optional[str] = None
+    ) -> Optional[List[str]]:
+        notice_dir = notice_dir if notice_dir else self.directory
+        if not os.path.exists(notice_dir):  # type: ignore
+            return None
+
+        notices = [
+            file
+            for file in os.listdir(notice_dir)
+            if os.path.isfile(os.path.join(notice_dir, file))  # type: ignore
+        ]
+        notices.sort()
+        return notices
+
+    def read(self) -> List[List[Optional[str]]]:
+        """Gets all the notice files currently saved.
+
+        :returns: List of notice file contents.
+        """
+        notice_directories = (
+            defaults.NOTICES_PERMANENT_DIRECTORY,
+            defaults.NOTICES_TEMPORARY_DIRECTORY,
+        )
+        file_notices = []
+        for notice_directory in notice_directories:
+            notices = self.get_all_notice_files(notice_directory)
+            notices = notices if notices is not None else []
+            for notice in notices:
+                label = notice.split("-")[1]
+                file_notices.append(
+                    [
+                        label,
+                        UAFile(
+                            name=notice,
+                            directory=notice_directory,  # type: ignore
+                        ).read(),
+                    ]
+                )
+        return file_notices
+
+
+_notice_cls = None
+
+
+def get_notice():
+    global _notice_cls
+    if _notice_cls is None:
+        _notice_cls = Notices()
+
+    return _notice_cls
+
+
+def add(root_mode: bool, notice_details: NoticeFileDetails, **kwargs):
+    notice = get_notice()
+    description = notice_details.message.format(**kwargs)
+    notice.add(root_mode, notice_details, description)
```

Comment:
I think the `format` call should probably happen in the `Notices.add` function instead of here